src/com/crossbowffs/quotelock/modules/wikiquote/WikiquoteQuoteModule.java

```diff
@@ -42,13 +42,13 @@ public QuoteData getQuote(Context context) throws IOException {
         }
         String quoteAllText = quoteAllTextMatcher.group(0).replaceAll("<.*?>", "");
 
-        Matcher quoteTextMatcher = Pattern.compile("^.*?(?=(\\s|)(——|--))").matcher(quoteAllText);
```

Comment:
Actually, why are you matching the text and the source with 2 separate regexes? Why not just use groups and a single regex?

Also, how about a fallback option? If the text and the source cannot be split, just put everything as the text and leave the source empty. At least that way users will get *something* (even though it's not perfect).

Pseudocode:
```
text, source = all.split()
if text.match and source.match:
   return (text, source)
else:
   return (all, "")
```